### ISSUE TITLE
Page collection shows no entries

### ISSUE THREAD

**Issue by mikemartin** [statamic/cms#1365]
Any ideas why my page collection entries are not displaying? I'm getting this console error:

Console error:
```
TypeError: Cannot read property 'length' of null
    at fn (Listing.vue?4dc8:10)
    at r (vue.common.prod.js:6)
    at a.yt [as _t] (vue.common.prod.js:6)
    at Table.vue?aba5:5
    at a.vt [as _l] (vue.common.prod.js:6)
    at a.<anonymous> (Table.vue?aba5:5)
    at a.e._render (vue.common.prod.js:6)
    at a.r (vue.common.prod.js:6)
    at pn.get (vue.common.prod.js:6)
    at new pn (vue.common.prod.js:6)
```

<img width="1439" alt="Screen Shot 2020-02-15 at 9 51 17 am" src="https://user-images.githubusercontent.com/414211/74574749-c5e6f800-4fd8-11ea-99c9-6679da200ccf.png">

I suspect it could be a route issue. When I delete the below `pages` structure and all pages except `home`, the page listing loads successfully.

/collections/pages.yaml
```
title: Pages
route: '{parent_uri}/{slug}'
blueprints:
  - default
  - news
structure: pages
revisions: false
```

/structures/pages.yaml
```
title: Pages
expects_root: true
root: home
tree:
  -
    entry: 040322da-43d5-4e8e-8183-260bce79c1a4
  -
    title: Insights
    children:
      -
        entry: 8355de08-3f1d-40d7-8f94-30b1213006a5
      -
        entry: 0bcbb4c8-d2b7-473d-9a67-2ee84569df4f
      -
        entry: a303e9f3-87e2-476b-8e56-435eb50b62d6
      -
        entry: 65e87b41-8d65-4012-a718-5e434f1eac71
  -
    title: Resources
    children:
      -
        entry: 4aa0bd47-9d67-4f52-b05a-4200027fd996
      -
        entry: 57769949-d779-4273-9962-765e3c37852a
        children:
          -
            entry: 52f0e7fd-020a-4746-87cf-3ba7a1dcd8bf
          -
            entry: ceb101aa-4d42-4d7a-bb82-fd2ddb066d8e
  -
    title: Company
    children:
      -
        entry: 53b37fbc-cc30-43c7-b78a-340b39f8ca26
      -
        title: Careers
        url: '/about-us#culture'
      -
        entry: 43e6e3ce-63e7-49b1-a5c5-1ff4196f932f
```

/collections/pages/newsrooms.md
```
---
title: Newsroom
updated_by: bf1b3343-ece9-442b-b4a6-104c0e54313a
updated_at: 1581479515
id: 43e6e3ce-63e7-49b1-a5c5-1ff4196f932f
blueprint: default
icon: 'assets/icons/newspaper.svg'
headline: "Latest news and upcoming events"
---
```

**Comment by jackmcdade** [comment]
Not quite sure, @mikemartin! It's hard to test with that data because the IDs won't line up if i copy and paste em locally. Is there a repo you could share access so we can take a peek? Also, are there any AJAX errors in the Network tab?

**Comment by mikemartin** [comment]
@jackmcdade I did some (sooo much) troubleshooting and narrowed down the issue to my Blueprint.  🎉

It seems the error is caused by an imported fieldset. When I remove the import, the pages load just fine. 

It doesn't look like the fields from the imported fieldset are the issue, because the error still occurs even if the **actions** fieldset has an empty fields array.

Blueprint
```
title: Default
sections:
  main:
    display: Main
    fields:
      -
        handle: headline
        field:
          character_limit: 0
          type: text
          localizable: false
          display: Headline
      -
        handle: summary
        field:
          type: bard
          localizable: false
          display: Summary
          listable: hidden
          buttons:
            - bold
            - italic
          save_html: false
          toolbar_mode: floating
          link_noopener: false
          link_noreferrer: false
          target_blank: false
          reading_time: false
          fullscreen: false
          allow_source: false
      -
        **import: actions**
      -
        handle: image
        field:
          container: assets
          mode: grid
          folder: /
          restrict: false
          allow_uploads: true
          max_files: 1
          type: assets
          localizable: false
          display: Image
      -
```

Fieldset:

```
title: Actions
fields:
  actions:
    fields:
      -
        handle: label
        field:
          character_limit: 0
          type: text
          localizable: false
          display: Label
      -
        handle: style
        field:
          options:
            link: Link
            primary: 'Primary Button'
            secondary: 'Secondary Button'
            dark: 'Dark Button'
          clearable: false
          multiple: false
          searchable: true
          taggable: false
          push_tags: false
          type: select
          localizable: false
          display: Style
      -
        handle: link
        field:
          character_limit: 0
          type: text
          localizable: false
          display: Link
    mode: table
    max_rows: 0
    min_rows: 0
    add_row: 'Add Action'
    reorderable: true
    type: grid
    instructions: 'Add buttons or links'
    localizable: false
    width: 100
    display: Actions

```

**Comment by mikemartin** [comment]
Solution:

Add a prefix to the imported action. 

```
-
    import: actions
    prefix: hero_
```

**Comment by jasonvarga** [comment]
I'm not 100% sure, but this seems structure-related.

This may be fixed in the next release along with all the other structure-related changes.

Please re-open if not.

**Comment by Turpoint** [comment]
I know this is closed, but I just had it too. For me, it got the error when I disabled JSON wrapping. Make sure you didn't add this line in the `boot` method of `AppServiceProvider`:

```
JsonResource::withoutWrapping();
```

**Comment by jasonvarga** [comment]
You cannot disable JSON wrapping at the moment. The whole CP is expecting it to be wrapped. Thanks for pointing that out though.

**Comment by jasonvarga** [comment]
It should have been fixed in 3.3.58 though.

**Comment by manogi** [comment]
I have looked and the error only appeared in the listings of my "Runway" (https://github.com/duncanmcclean/runway) models, not in the collection listings. So this is probably still fixed in statamic core, and I will maybe move this discussion to the runway repository.
